### ISSUE TITLE
Problem with segmentation my dataset in DeepLab 

### ISSUE THREAD

**Issue by rafaelmarconiramos** [tensorflow/models#8253]
I adapted the "build_convert" of the PASCAL Dataset to convert my dataset to execute the DeepLab, but all my segmentation tests generate a black image prediction. 

My dataset has images with 140x140 px. (see an example above:)

Original Image
![211789](https://user-images.githubusercontent.com/18166522/75722022-256f2400-5cb8-11ea-9308-0c7637d37cb5.png)

Original Mask
![211789_OriginalMask](https://user-images.githubusercontent.com/18166522/75721512-3d927380-5cb7-11ea-866d-1e16364d073b.png)

Converted Mask
![211789_ConvertedMask](https://user-images.githubusercontent.com/18166522/75721539-4b47f900-5cb7-11ea-9657-8644aa8eb335.png)

Prediction Image (complete black, all prediction pixels are background)
![211789_Prediction](https://user-images.githubusercontent.com/18166522/75721593-69155e00-5cb7-11ea-8580-61f163bf611c.png)

What am I doing wrong?
Could anybody help me with this problem?

My step-by-step:

1) Convert the segmentation masks

python ./remove_gt_colormap.py \
  --original_gt_folder="${SEG_FOLDER}" \
  --output_dir="${SEMANTIC_SEG_FOLDER}"
SEG_FOLDER -> the original mask images

2) Create the tfrecord files
(I reused the  PASCAL example)

python ./build_voc2012_data.py \
  --image_folder="${IMAGE_FOLDER}" \
  --semantic_segmentation_folder="${SEMANTIC_SEG_FOLDER}" \
  --list_folder="${LIST_FOLDER}" \
  --output_dir="${OUTPUT_DIR}"

3) To train

NUM_ITERATIONS=100
python "${WORK_DIR}"/train.py \
  --logtostderr \
  --train_split="val" \
  --model_variant="${modelo}" \
  --atrous_rates=12 --atrous_rates=24 --atrous_rates=36 --output_stride=8 \
  --decoder_output_stride=4 \
  --train_crop_size="513,513" \
  --train_batch_size=4 \
  --training_number_of_steps="${NUM_ITERATIONS}" \
  --train_logdir="${TRAIN_LOGDIR}" \
  --dataset_dir="${PASCAL_DATASET}"

4) to eval
python "${WORK_DIR}"/vis.py \
  --logtostderr \
  --vis_split="val" \
  --model_variant="${modelo}" \
  --atrous_rates=12 --atrous_rates=24 --atrous_rates=36 --output_stride=8 --decoder_output_stride=4 \
  --vis_crop_size="513,513" \
  --checkpoint_dir="${TRAIN_LOGDIR}" \
  --vis_logdir="${VIS_LOGDIR}" \
  --dataset_dir="${PASCAL_DATASET}" \
  --max_number_of_iterations=1

Thanks

Thanks

**Comment by hmaresearch2** [comment]
We are having similar issues as above. Training loss seems to converge but "vis" is generating blank images. Followed all steps as mentioned above and here are more details

Training Image size: 600x800x3
Annotation: 600x800
Classes: 2 ( background + 1 class)

pertained model: deeplabv3_mnv2_pascal_train_aug/model.ckpt-30000

Configuration parameters
!python "models/research/deeplab/train.py" \
  --logtostderr \
  --train_split="trainval" \
  --model_variant="mobilenet_v2" \
  --output_stride=16 \
  --train_crop_size="321,321" \
  --train_batch_size=4 \
  --training_number_of_steps=2000 \
  --fine_tune_batch_norm=true \
  --tf_initial_checkpoint="${DATADIR}/models/research/deeplab/datasets/CustomLabel/pretrained/deeplabv3_mnv2_pascal_train_aug/model.ckpt-30000" \
  --train_logdir="${DATADIR}/models/research/deeplab/datasets/CarWindow2/training7" \
  --initialize_last_layer=False \
  --last_layers_contain_logits_only=True \
  --num_classes=2 \
  --dataset="carwindow" \
  --dataset_dir="${DATADIR}models/research/deeplab/datasets/CarWindow2/tfrecord2"


# Visualize the results.
!python "models/research/deeplab/vis.py" \
  --logtostderr \
  --vis_split="val" \
  --model_variant="mobilenet_v2" \
  --vis_crop_size="600,800" \
  --checkpoint_dir="${DATADIR}/models/research/deeplab/datasets/CarWindow2/training7" \
  --vis_logdir="${DATADIR}/models/research/deeplab/datasets/CarWindow2/vis7" \
  --dataset_dir="${DATADIR}/models/research/deeplab/datasets/CarWindow2/tfrecord2" \
  --dataset="customlabel" \
  --num_classes=2 \
  --max_number_of_iterations=1
Made changes to code in below locations
data_generator.py
_CUSTOMLABEL_INFORMATION = DatasetDescriptor(
    splits_to_sizes={
        'train': 39,  # num of samples in images/training
        'val': 6,  # num of samples in images/validation
        'trainval': 45,
    },
    num_classes=2,
    ignore_label=255,
)

train_utils.py
loss_weight=[0.1, 0.9] 

Any suggestions on how to get this fixed

**Comment by LightingX** [comment]
I have met the same problem, have you solved it?

**Comment by LightingX** [comment]
I have met the same problem, have you solved it? I tried to tune the weight with [0.1, 0.9] by the param loss_weights when training, but got the same result.

**Comment by rafaelmarconiramos** [comment]
Hello @LightingX and @hmaresearch2, Did you find any solution for the problem?
Thanks

**Comment by vcjmhg** [comment]
Hi,@LightingX  @hmaresearch2 @rafaelmarconiramos , Did you find any solution for the problem?
Thanks

**Comment by edsumpena** [comment]
@rafaelmarconiramos 
1. Train with more steps/iterations (100 is too little, 2000+ as least)
2. Load a pre-trained model checkpoint using --tf_initial_checkpoint="model.ckpt" since you are training with a low number of steps.

Hope this helps.

**Comment by lavender-ling** [comment]
> I have met the same problem, have you solved it? I tried to tune the weight with [0.1, 0.9] by the param loss_weights when training, but got the same result.

你解决了这个问题吗，谢谢

**Comment by snooplsm** [comment]
One problem I see is your model variant is mobilenetv2 and you have --atrous_rates=12 --atrous_rates=24 --atrous_rates=36 defined.  You need to remove the atrous_rates since the train.py says that variable should be None.
